src/Salt/SiteBundle/Controller/CommentsController.php

```diff
@@ -159,33 +160,92 @@ public function downvoteAction(Comment $comment, UserInterface $user)
      * @param Request $request
      *
      * @return Response
+     *
+     * @Security("is_granted('comment_view')")
      */
     public function exportCommentAction(string $itemType, int $itemId)
     {
-        $url = ($itemType === 'item') ? $this->generateUrl('doc_tree_item_view', ['id' => $itemId], UrlGeneratorInterface::ABSOLUTE_URL) : $this->generateUrl('doc_tree_view', ['slug' => $itemId], UrlGeneratorInterface::ABSOLUTE_URL);
+        $childIds = [];
+        $child_comment_rows = [];
         $repo = $this->getDoctrine()->getManager()->getRepository(Comment::class);
-        $headers = ['Framework Name', 'Node Address', ($itemType === 'item') ? 'HumanCodingScheme' : null, 'User', 'Organization', 'Comment', 'Created Date', 'Updated Date'];
-        $rows[] = implode(',', array_filter($headers));
-        $comment_data = $repo->findBy([$itemType => $itemId]);
+        $lsItemRepo = $this->getDoctrine()->getManager()->getRepository(LsItem::class);
+        $headers = ['Framework Name', 'Node Address', 'HumanCodingScheme', 'User', 'Organization', 'Comment', 'Created Date', 'Updated Date'];
+        switch ($itemType)
+        {
+            case 'document':
+                $comment_data = $repo->findBy([$itemType => $itemId]);
+                $comment_rows = $this->csvArray($comment_data, $itemType);
+                foreach ($comment_rows as $row)
+                {
+                    $rows[] = $row;
+                }
+                $lsDoc = $this->getDoctrine()->getManager()->getRepository(LsDoc::class)->find($itemId);
+                $lsDocChilds = $lsDoc->getLsItems();
+                foreach ($lsDocChilds as $lsDocChild){
+                    $childIds[] = $lsDocChild->getId();
+                }
+                break;
+
+            case 'item':
+                $lsItem = $lsItemRepo->findOneById($itemId);
+                $lsItem->getDescendantIds($childIds);
+                $childIds[] = $itemId;
+                break;
+        }
+        $comment_data = $repo->findBy(['item' => $childIds]);
+        $comment_rows = $this->csvArray($comment_data, 'item');
+        foreach ($comment_rows as $child_row)
+        {
+            $rows[] = $child_row;
+        }
+        $response = new StreamedResponse();
+        $response->setCallback(function () use($rows) {
```

Comment:
Most of the above can be put into the callback function, doing the `fputcsv` directly and eliminating the need for the `$rows` array.